packages/api/server/ws.mts

```diff
@@ -306,16 +282,109 @@ async function cellUpdate(payload: CellUpdatePayloadType) {
     wss.broadcast(`session:${session.id}`, 'cell:updated', {
       cell: findCell(session, payload.cellId),
     });
+
+    return;
+  }
+
+  if (
+    session.metadata.language === 'typescript' &&
+    result.cell.type === 'code' &&
+    tsservers.has(session.id)
+  ) {
+    const cell = result.cell;
+    const tsserver = tsservers.get(session.id);
+
+    const file = Path.join(session.dir, cell.filename);
+
+    // To update a file in tsserver, close and reopen it. I assume performance of
+    // this implementation is worse than calculating diffs and using `change` command
```

Comment:
I've heard that diffs are significantly faster, but if this feels fast enough, seems good enough for now. It also has the benefit of not being stateful and therefore not having TSServer state bugs (I see these in NeoVim here and there and need to kick the LSP)

Reply:
For large projects with lots of big open files, I imagine that's true. For a couple files with small amounts of code in them, I imagine it's not that different. We give the source directly, so I don't expect tsserver to interact with the disk. If it's not going to disk, it's all in-memory already. The alternative requires calculating diffs, line numbers, offsets, extracting the change, and then passing that to tsserver for it to then do some more computation.

In either case, this won't have end-user impact and this is far simpler without extra dependencies for diffing. Happy to change in the future if this becomes a problem